tests/cephfs/cephfs_scale/max_snapshots.py

```diff
@@ -0,0 +1,153 @@
+       [--pool_layout <data_pool_name>] [--uid <uid>] [--gid <gid>] [--mode <octal_mode>]  [--namespace-isolated]
+       Ex: ceph fs subvolume create cephfs subvol_max_snap --size 5368706371 --group_name subvolgroup_1
+    4. Create Data on the subvolume
+        Ex:  python3 /home/cephuser/smallfile/smallfile_cli.py --operation create --threads 10 --file-size 400 --files
+            100 --files-per-dir 10 --dirs-per-dir 2 --top /mnt/cephfs_fuse1baxgbpaia_1/
+
+    Test Script Flow :
+    1. We will set max snapshots to maximum allowed value 4096.
+    2. We add data based on max available data divided by 4096 .
+    3. For every snapshot we are going to write new file until 4096
+
+    Clean up:
+    1. Deletes all the snapshots created
+    2. Deletes snapshot and subvolume created.
+    """
+    try:
+        fs_util = FsUtils(ceph_cluster)
+        config = kw.get("config")
+        clients = ceph_cluster.get_ceph_objects("client")
+        build = config.get("build", config.get("rhbuild"))
+        fs_util.prepare_clients(clients, build)
+        fs_util.auth_list(clients)
+        max_snap = 4096
+        log.info("checking Pre-requisites")
+        if len(clients) < 1:
+            log.info(
+                f"This test requires minimum 1 client nodes.This has only {len(clients)} clients"
+            )
+            return 1
+        default_fs = "cephfs"
+        mounting_dir = "".join(
+            random.choice(string.ascii_lowercase + string.digits)
+            for _ in list(range(10))
+        )
+        client1 = clients[0]
+        fs_details = fs_util.get_fs_info(client1)
+        if not fs_details:
+            fs_util.create_fs(client1, "cephfs")
+        subvolume = {
+            "vol_name": default_fs,
+            "subvol_name": "subvol_max_snap",
+        }
+        fs_util.create_subvolume(client1, **subvolume)
+        log.info("Get the path of sub volume")
+        subvol_path, rc = client1.exec_command(
+            sudo=True,
+            cmd=f"ceph fs subvolume getpath {default_fs} subvol_max_snap",
+        )
+        client1.exec_command(
+            sudo=True,
+            cmd=f"ceph config set mds mds_max_snaps_per_dir {max_snap}",
+        )
+        kernel_mounting_dir_1 = f"/mnt/cephfs_kernel{mounting_dir}_1/"
+        mon_node_ips = fs_util.get_mon_node_ips()
+        fs_util.kernel_mount(
+            [clients[0]],
+            kernel_mounting_dir_1,
+            ",".join(mon_node_ips),
+            sub_dir=f"{subvol_path.strip()}",
+        )
+        snapshot_list = [
+            {
+                "vol_name": default_fs,
+                "subvol_name": "subvol_max_snap",
+                "snap_name": f"snap_limit_{x}",
+            }
+            for x in range(1, max_snap + 1)
+        ]
+        available_space = int(get_available_space(client1, default_fs))
+        log.info(available_space)
+        data_size_iter = int(available_space / max_snap)
+        bs = int(data_size_iter / 10000)
```

Comment:
I see "10000" for bs  & in count for dd is "1000" . Is this intentional?

Reply:
Also could we decrease the "bs" so we can write 4096 files. As we are seeing failures to write 4096 files

Reply:
Those values are intended to reduce each size of the object written into ceph.
with these values, we do not see any issues while running dd commands